### ISSUE TITLE
Allow trailing commas in settings JSON

### ISSUE THREAD

**Issue by alexreg** [Microsoft/vscode#26830]
- VSCode Version: 1.12.2
- OS Version: OS X 10.12.4

Trailing commas aren't allowed in settings JSON, it seems. Could we allow this in the future please?

**Comment by babakness** [comment]
JSON can't have trailing commas. https://stackoverflow.com/questions/201782/can-you-use-a-trailing-comma-in-a-json-object

`npm` complains about them all the time. 

Then again, I think even comments are non-standard even though I see them in the settings JSON. While this makes it incompatible with the standard, since we are already not following, I suppose it would make sense. Basically treat it as a Javascript object instead of JSON.

**Comment by alexreg** [comment]
True, although apparently JSONv5 can(?) Anyway, since the purpose of JSON here is only for settings, I don't think it's too bad to add a minor 'proprietary' extension. Your point about comments too also adds justification.

**Comment by sandy081** [comment]
@alexreg Trailing comma in settings.json is ignored while reading it. I know the validation is still shown but this is a regular JSON validator. 

@aeschli Can we also ignore validations?

**Comment by aeschli** [comment]
We should allow trailing commas (and comments) only on our settings files. Currently, the language JSON language support doesn't know which files these are. So it's not an easy fix. I'd like to keep the trailing comma error until then.

**Comment by DamianPereira** [comment]
Some extensions seem to use JSON.parse to check settings. I found an error with spellRight, which was pretty hard to debug from a user POV, I had to get into the dev tools and check which line failed, only to see that JSON.parse was causing an exception because of the trailing comma. 

I think VSCode should throw an error or warning if the settings.json file is saved with a trailing comma. At least if JSON.parse is the preffered method to read settings on extensions code. Or is there an API to read them and direct file access is not recommended?
